src/sync/app.cpp

```diff
@@ -81,11 +93,425 @@ static Optional<AppError> check_for_errors(const Response& response) {
+    if (!this->parent) {
+        return completion_block(AppError(make_custom_error_code(0), "UsernamePasswordProviderClient parent is null"));
+    }
+
+    std::string route = util::format("%1/providers/%2/reset", parent->m_auth_route, m_provider_key);
+    
+    auto handler = [completion_block](const Response& response) {
+        if (auto error = check_for_errors(response)) {
+            return completion_block(error);
+        } else {
+            return completion_block({});
+        }
+    };
+    
+    nlohmann::json body = {
+        { "password", password },
+        { "token", token },
+        { "token_id", token_id }
+    };
+
+    parent->m_config.transport_generator()->send_request_to_server({
+        HttpMethod::post,
+        route,
+        parent->m_request_timeout_ms,
+        get_request_headers(false),
+        body.dump()
+    }, handler);
+}
+
+void App::UsernamePasswordProviderClient::call_reset_password_function(const std::string& email,
+                                                                       const std::string& password,
+                                                                       const std::string& args,
+                                                                       std::function<void(Optional<AppError>)> completion_block) {
+    if (!this->parent) {
+        return completion_block(AppError(make_custom_error_code(0), "UsernamePasswordProviderClient parent is null"));
+    }
+
+    std::string route = util::format("%1/providers/%2/reset/call", parent->m_auth_route, m_provider_key);
+
+    auto handler = [completion_block](const Response& response) {
+        if (auto error = check_for_errors(response)) {
+            return completion_block(error);
+        } else {
+            return completion_block({});
+        }
+    };
+
+    nlohmann::json body = {
+        { "email", email },
+        { "password", password },
+        { "arguments", nlohmann::json::parse(args) }
+    };
+    
+    parent->m_config.transport_generator()->send_request_to_server({
+        HttpMethod::post,
+        route,
+        parent->m_request_timeout_ms,
+        get_request_headers(false),
+        body.dump()
+    }, handler);
+}
+
+// MARK: - UserAPIKeyProviderClient
+
+ void App::UserAPIKeyProviderClient::create_api_key(const std::string &name,
+                                                   std::function<void (Optional<UserAPIKey>, Optional<AppError>)> completion_block) {
+    if (!this->parent) {
+        return completion_block(Optional<UserAPIKey>(), AppError(make_custom_error_code(0), "UserAPIKeyProviderClient parent is null"));
+    }
+
+    std::string route = util::format("%1/auth/%2", parent->m_base_route, m_provider_key);
+    
+    auto handler = [completion_block](const Response& response) {
+
+        if (auto error = check_for_errors(response)) {
+            return completion_block(Optional<UserAPIKey>(), error);
```

Comment:
`{}` is used in the previous functions for this. I don't think it really matters which is used, but it'd be nice to be consistent.

Reply:
I agree about consistency, I have changed to using `{}`